api/testworkflows/v1/testworkflowexecution_type.go

```diff
@@ -0,0 +1,233 @@
+limitations under the License.
+*/
+
+package v1
+
+import (
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/intstr"
+)
+
+// TestWorkflowExecutionSpec defines the desired state of TestWorkflowExecution
+type TestWorkflowExecutionSpec struct {
+	// INSERT ADDITIONAL SPEC FIELDS - desired state of cluster
+	// Important: Run "make" to regenerate code after modifying this file
+
+	TestWorkflow     *corev1.LocalObjectReference  `json:"testWorkflow" expr:"include"`
+	ExecutionRequest *TestWorkflowExecutionRequest `json:"executionRequest,omitempty" expr:"include"`
+}
+
+// TestWorkflowExecutionRequest contains TestWorkflow execution parameters
+type TestWorkflowExecutionRequest struct {
+	// custom execution name
+	Name   string                        `json:"name,omitempty" expr:"template"`
+	Config map[string]intstr.IntOrString `json:"config,omitempty" expr:"template"`
+	// test workflow execution name started the test workflow execution
+	TestWorkflowExecutionName string `json:"testWorkflowExecutionName,omitempty" expr:"template"`
+}
+
+// TestWorkflowExecutionStatus defines the observed state of TestWorkflowExecution
+type TestWorkflowExecutionStatus struct {
+	// INSERT ADDITIONAL STATUS FIELD - define observed state of cluster
+	// Important: Run "make" to regenerate code after modifying this file
+	LatestExecution *TestWorkflowExecutionDetails `json:"latestExecution,omitempty" expr:"include"`
+	// test workflow execution generation
+	Generation int64 `json:"generation,omitempty" expr:"template"`
+}
+
+// TestWorkflowExecutionDetails contains the details of TestWorkflowExecution
+type TestWorkflowExecutionDetails struct {
+	// unique execution identifier
+	Id string `json:"id" expr:"template"`
+	// execution name
+	Name string `json:"name" expr:"template"`
+	// execution namespace
+	Namespace string `json:"namespace,omitempty" expr:"template"`
+	// sequence number for the execution
+	Number int32 `json:"number,omitempty" expr:"template"`
+	// when the execution has been scheduled to run
+	ScheduledAt metav1.Time `json:"scheduledAt,omitempty" expr:"template"`
+	// when the execution result's status has changed last time (queued, passed, failed)
+	StatusAt metav1.Time `json:"statusAt,omitempty" expr:"template"`
+	// structured tree of steps
+	Signature []TestWorkflowSignature `json:"signature,omitempty" expr:"include"`
+	Result    *TestWorkflowResult     `json:"result,omitempty" expr:"include"`
+	// additional information from the steps, like referenced executed tests or artifacts
+	Output []TestWorkflowOutput `json:"output,omitempty" expr:"include"`
+	// generated reports from the steps, like junit
+	Reports          []TestWorkflowReport `json:"reports,omitempty" expr:"include"`
+	Workflow         *TestWorkflow        `json:"workflow" expr:"include"`
+	ResolvedWorkflow *TestWorkflow        `json:"resolvedWorkflow,omitempty" expr:"include"`
```

Comment:
There is a chance that you may need to mark them as schemaless - the TestWorkflow CRD is near Kubernetes constraints, while there are two of them; Unfortunately, Kubernetes doesn't support `$ref` yet 🙁